### ISSUE TITLE
Cargo stuck on "Updating crates.io index" when run from docker on DigitalOcean droplet

### ISSUE THREAD

**Issue by nseymoursmith** [nseymoursmith/waterlogged-prototype#4]
Also gets stuck here when running on my machine (ubuntu 22). Not cool cargo, not cool.

Anyway, it looks like other people have seen this and added some wizardry to the build definitions to fix it:

https://github.com/rust-lang/docker-rust/issues/111
https://github.com/rust-lang/cargo/issues/10781
https://github.com/rust-lang/rust/issues/64248
https://stackoverflow.com/questions/67001033/skip-updating-crates-io-index-when-using-cargo-run

To make this work, we should figure out which of these, if any, is the right bit of wizardry.

**Comment by nseymoursmith** [comment]
@KevNickells @KevT-plan (which of these should I use), just alerting you to the fact that this is blocking the piece of work that I said I'd do to unblock you!
I'll get round to it soon.

**Comment by KevNickells** [comment]
> @KevNickells @KevT-plan (which of these should I use), just alerting you to the fact that this is blocking the piece of work that I said I'd do to unblock you! I'll get round to it soon.

Current account - @KevNickells - thanks!

**Comment by nrs23** [comment]
OK So I've tried a series of solutions and I have thoughts:
First two solutions come from: https://github.com/rust-lang/docker-rust/issues/111#issuecomment-1258513924

I tried using the spare build index option suggestion, but apparently now thats the default so this is not working.

I then tried the extra steps to build the index in the docker image before compiling the code, it still hung.

The last solution that could be tried is explained here: https://github.com/rust-lang/docker-rust/issues/111#issuecomment-1366015055

and it essentially amounts to building the index on your local machine and sharing that with the docker image at build time.

I don't really think that that solution is very satisfactory as it undercuts the benefit of using docker in the first place by relying on your local environment. 

SO my feeling is that maybe for now docker isn't the right solution for using rust since

a) running a similar docker command on your machine and mine is producing different results
b) the results on my machine are unusable without putting in a bunch of extra work and/or relying on local environment

And the question then is:
- should we just trust rust to compile correctly across our different platforms (ditch docker?) 
- should we ditch rust for something else?
- Am I jumping the gun?

Thoughts appreciated @KevNickells

**Comment by nrs23** [comment]
Latest commit to `deploy-without-docker` branch is a good stopping point to consider the 'ditch docker' approach:
https://github.com/nseymoursmith/waterlogged-prototype/commit/d8996fd8a5777795f8746d6b5d2b1734121d0de1

`deploy-rust.sh` script tested on the DO droplet and working.

**Comment by nseymoursmith** [comment]
I've broken out the 'deploy without docker' bit to its own issue (#6) because there's potentially still other solutions to this issue (though I'm not too keen on digging further personally).
